platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java

```diff
@@ -128,212 +46,21 @@ public InOrderLinker(@NonNull final PlatformContext platformContext) {
-    }
-
-    /**
-     * This method is called when this data structure stops tracking an event because it has become ancient.
-     * @param event the event that has become ancient
-     */
-    protected void eventHasBecomeAncient(@NonNull final EventImpl event) {
-        // Implement this if extra action is needed.
-    }
-
-    /**
-     * Find the correct parent to link to a child. If a parent should not be linked, null is returned.
-     * <p>
-     * A parent should not be linked if any of the following are true:
-     * <ul>
-     *     <li>The parent is ancient</li>
-     *     <li>The parent's generation does not match the generation claimed by the child event</li>
-     *     <li>The parent's birthRound does not match the claimed birthRound by the child event</li>
-     *     <li>The parent's time created is greater than or equal to the child's time created</li>
-     * </ul>
-     *
-     * @param child            the child event
-     * @param parentDescriptor the event descriptor for the claimed parent
-     * @return the parent to link, or null if no parent should be linked
-     */
-    @Nullable
-    private EventImpl getParentToLink(
-            @NonNull final GossipEvent child, @Nullable final EventDescriptor parentDescriptor) {
-
-        if (parentDescriptor == null) {
-            // There is no claimed parent for linking.
-            return null;
-        }
-
-        if (eventWindow.isAncient(parentDescriptor)) {
-            // ancient parents don't need to be linked
-            return null;
-        }
-
-        final EventImpl candidateParent = parentHashMap.get(parentDescriptor.getHash());
-        if (candidateParent == null) {
-            childHasMissingParent(child, parentDescriptor);
-            return null;
-        }
-
-        if (candidateParent.getGeneration() != parentDescriptor.getGeneration()) {
-            parentHasIncorrectGeneration(child, parentDescriptor, candidateParent);
-            return null;
-        }
-
-        if (candidateParent.getBirthRound() != parentDescriptor.getBirthRound()) {
-            parentHasIncorrectBirthRound(child, parentDescriptor, candidateParent);
-            return null;
-        }
-
-        final Instant parentTimeCreated =
-                candidateParent.getBaseEvent().getHashedData().getTimeCreated();
-        final Instant childTimeCreated = child.getHashedData().getTimeCreated();
-
-        // only do this check for self parent, since the event creator doesn't consider other parent creation time
-        // when deciding on the event creation time
-        if (parentDescriptor.getCreator().equals(child.getDescriptor().getCreator())
-                && parentTimeCreated.compareTo(childTimeCreated) >= 0) {
-
-            childTimeIsNotAfterSelfParentTime(child, candidateParent, parentTimeCreated, childTimeCreated);
-
-            return null;
-        }
-
-        return candidateParent;
-    }
+    void clear(@NonNull final NoInput ignored);
```

Comment:
optional: add wire annotation

Reply:
The default label for an input wire is the name of the method. In this case, the name of the method is the name I want to have for this wire.